packages/cli-plugin-metro/src/tools/KeyPressHandler.ts

```diff
@@ -0,0 +1,73 @@
+import {CLIError, logger} from '@react-native-community/cli-tools';
+
+const CTRL_C = '\u0003';
+
+/** An abstract key stroke interceptor. */
+export class KeyPressHandler {
+  private isInterceptingKeyStrokes = false;
+  private isHandlingKeyPress = false;
+
+  constructor(public onPress: (key: string) => Promise<any>) {}
```

Comment:
I think we're don't intend to return anything from this fn and if it must be either sync or async, let's state that in the type
```suggestion
  constructor(public onPress: (key: string) => Promise<void> | void) {}
```